CppCoreGuidelines.md

```diff
@@ -7853,9 +7933,13 @@ One example is a performance-critical AST hierarchy in a compiler or language an
 New derived classes are not added every year and only by library implementers.
 However, misuses are (or at least have been) far more common.
 
+##### Note
+
+[A class with a public non-virtual destructor should be final](#Rc-dtor-final). So, the use of non-virtual interfaces may lead to some restrictions.
+
 ##### Enforcement
 
-Flag uses of `final` on classes.
+Flag uses of `final` on classes with virtual destructor.
```

Comment:
Only with virtual destructor or with any virtual function?

Reply:
It does not connected to virtual functions. The only destruction process make sense. The classes with virtual destructors could be safely destroyed by pointer to the base class. So, it isn't useful to prevent their inheritance. Having an virtual functions in a class does not provides virtual destructor by default.